extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -125,10 +128,8 @@ public <T, R> List<R> executeSelect(
         ResultSet resultSet = statement.executeQuery(query)) {
       List<R> resultList = new ArrayList<>();
       while (resultSet.next() && resultList.size() < limit) {
-        Converter<T> object =
-            (Converter<T>)
-                entityClass.getDeclaredConstructor().newInstance(); // Create a new instance
-        R entity = transformer.apply(object.fromResultSet(resultSet));
+        Converter<T> model = (Converter<T>) createModelInstance(entityClass);
```

Comment:
How about changing `entityClass` to `Function<ResultSet, R> converter` and removing `transformer`.

Then:
```
public static ModelEntity fromResultSet0(ResultSet r)  {
    return ModelEntity.builder()
        ....
```

End result no reflection, not if/else, one less method param, simpler entity classes.

Reply:
+1. It's a good idea to combine `toEntity()` and `fromResultSet()` here.